File: poc/3D_model_construction/src/Point/Point.cpp
#include "Point.h"
#include <math.h>

Point::Point(float x1, float y1, float z1) {
    x = x1;
    y = y1;
    z = z1;
}

float Point::distSqr(Point a, Point b) {
    return pow(b.x-a.x, 2) + pow(b.y-a.y, 2) + pow(b.z-a.z, 2);
}

float Point::dist(Point a, Point b) {
    return sqrt(Point::distSqr(a, b));
}

File: wasm/src/model_construction/point.rs
#[derive(Debug)]
pub struct Point {
	pub x: f32,
	pub y: f32,
	pub z: f32,
}

impl Point {
	// Constructor is not necessary

	pub fn dist_sqr(a: &Point, b: &Point) -> f32 {
		(b.x - a.x).powi(2) + (b.y - a.y).powi(2) + (b.z - a.z).powi(2)
	}

	pub fn dist(a: &Point, b: &Point) -> f32 {
		f32::sqrt(Point::dist_sqr(a, b))
	}

	pub fn xy_dist_sqr(a: &Point, b: &Point) -> f32 {
		(b.x - a.x).powi(2) + (b.y - a.y).powi(2)
	}

	pub fn xy_dist(a: &Point, b: &Point) -> f32 {
		f32::sqrt(Point::xy_dist_sqr(a, b))
	}
}
